src/permissions/NativeTokenRollingSpendLimit/NativeTokenRollingSpendLimitPermission.sol

```diff
@@ -106,72 +96,79 @@ contract NativeTokenRollingSpendLimitPermission is IPermissionContract {
 
                 // check withdraw is native token
                 if (withdraw.asset != address(0)) revert InvalidWithdrawAsset();
-                // do not need to accrue spendValue because withdrawn value will be spent in other calls
-            } else if (selector == IMagicSpend.withdrawGasExcess.selector) {
-                // ok
-            } else if (selector == NativeTokenRollingSpendLimitPermission.assertSpend.selector) {
-                // check call index last call
-                if (i < callsLen - 1) revert MustAssertSpendLastCall();
-
-                // check call target is this contract
-                if (call.target != address(this)) revert UserOperationUtils.TargetNotAllowed();
-
-                // prepare expected call data for `assertSpend`
-                bytes memory assertSpendData = abi.encodeWithSelector(
-                    NativeTokenRollingSpendLimitPermission.assertSpend.selector,
-                    permissionHash,
-                    spendPeriodDuration,
-                    spendPeriodLimit,
-                    spendValue
-                );
-
-                // check call data matches prepared assertSpend arguments
-                if (keccak256(calls[callsLen - 1].data) != keccak256(assertSpendData)) {
-                    revert UserOperationUtils.ArgumentsNotAllowed();
-                }
-
-                // mark assertSpend callback
-                makesAssertSpendCallback = true;
-            } else {
+                // do not need to accrue callsSpend because withdrawn value will be spent in other calls
+            } else if (selector != IMagicSpend.withdrawGasExcess.selector) {
                 revert UserOperationUtils.SelectorNotAllowed();
```

Comment:
should we still allow users to withdraw here rather than reverting? It seems like the gas accounting (if magicspend is used as a paymaster) is moved into the `assertSpend` logic below, but could we still let users call `withdrawGasExcess` and then any value that the user spends in subsequent calls should still be caught. If we didn't allow this selector, I think it might disable to withdraw from magicspend during the SCW's execution time